File: countingvalleys.js
function processData(input) {
     var level = 0; 
     var valleys = 0;
     var belowSea = false;
    
     for(i = 0; i < input.length; i++) {
         var terrain = input[i];
         if (terrain === "U") {
            level++   
         } else if (terrain === "D") {
             level--
         
             }
           if (belowSea == false && level < 0) {
                valleys++;
                belowSea = true;
            }
            
           if(level >= 0) {
                belowSea = false;
        }
     }
        console.log(valleys)
    }

File: countingvalleys.rb
def countValley(input)

level = 0
valleys = 0
belowSea = 0 

(0..input.length).each do |i|
    terrain = input[i]
    if terrain == "U"
        level += 1
    elsif terrain == "D"
        level -= 1
    end

    if belowSea == false and level < 0
        valleys -= 1
        belowwSea = true
    end
    
    if level >= 0
        belowSea = false
    end
end

puts valleys

end